### PR TITLE
Authors pages feature

### PR DESCRIPTION
**Pull request by taylorrf**
In order to have a good version of authors pages, it’s:

* Add to books a cover image and an author
* Add description to author.
* Design of authors and author pages
* Some design of books and book page.

Closing the cards:
https://trello.com/c/AfGBgf9I/15-list-author-name-bio-etc
https://trello.com/c/COkdrf9J/14-for-each-author-show-name-and-sample-co
vers-of-up-to-5-of-their-books
https://trello.com/c/YzW65X3z/16-list-author-books-similar-to-display-on
-list-page

### CODE REVIEW

File: app/assets/stylesheets/layout/_authorslist.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
+.authorslist {
```
Comment (andyw8) [inline app/assets/stylesheets/layout/_authorslist.scss]: Thoughts on separating the words with a hyphen or underscore for easier reading?

File: app/models/author.rb [hunk 2]
```diff
@@ -1,3 +1,8 @@
 class Author < ActiveRecord::Base
   validates :name, presence: true
+  has_many :books
+
+  def popular_books
```
Comment (andyw8) [inline app/models/author.rb]: It feels to me that the number displayed should be a responsibility of the presentation level. Thoughts on passing in the limit as a value from the view? (If the design needed changed to show more or less books, then the Rails model itself shouldn't have to change).

File: config/initializers/kaminari_config.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 Kaminari.configure do |config|
-  config.default_per_page = 20
+  config.default_per_page = 10
```
Comment (andyw8) [inline config/initializers/kaminari_config.rb]: Why has this changed? (The challenge instruction specify 20 per page).

File: spec/views/books/show.html.erb_spec.rb [hunk 4]
```diff
@@ -18,7 +18,52 @@
...
+      book = build_stubbed(:book, cover: "book_cover.jpg")
+
+      render template: "books/show", locals: { book: book }
+
+      expect(rendered).to have_cover("book_cover.jpg")
+    end
+  end
+
+  context "when the book has no a cover sample" do
+    it "don't display the cover sample" do
```
Comment (andyw8) [inline spec/views/books/show.html.erb_spec.rb]: [grammar] `it "doesn't display...`

File: spec/views/authors/show.html.erb_spec.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      create(:book, author: author, title: "Legendary Ruby")
+
+      render template: "authors/show", locals: { author: author }
+
+      expect(rendered).to have_css("a", text: "Legendary Ruby")
+    end
+  end
+
+  context "when the author hasn't books" do
+    it "only displays their basic infos" do
```
Comment (andyw8) [inline spec/views/authors/show.html.erb_spec.rb]: [grammar] `info` (or `information`)

File: spec/views/authors/show.html.erb_spec.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
+require "rails_helper"
+
+RSpec.describe "authors/show" do
+  context "when the author has a books" do
```
Comment (andyw8) [inline spec/views/authors/show.html.erb_spec.rb]: [grammar] `has books`

File: spec/views/authors/show.html.erb_spec.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    it "displays their books" do
+      author = create(:author, name: "Aaron")
```
Comment (andyw8) [inline spec/views/authors/show.html.erb_spec.rb]: For a fast test suite it's good if the view specs don't hit the database. What do you think about about stubbing the association with something like:
```ruby
      author = build_stubbed(:author, name: "Aaron")
      book = build_stubbed(:book, author: author, title: "Legendary Ruby")
      allow(author).to receive(:books).and_return([book])

```

File: app/assets/stylesheets/layout/_authorslist.scss (hunk 1 above)
Comment (taylorrf) [inline app/assets/stylesheets/layout/_authorslist.scss]: Good idea

File: app/models/author.rb (hunk 2 above)
Comment (taylorrf) [inline app/models/author.rb]: I think is a good idea to extract this number as a default (like `popular_books(limit=5)`) and allow the views layer to ask for more when necessary but without delegate it as a view responsibility, what you think?

Also, I see it as a WIP method and in order to really get the "popular books" I believe we should calculate the ratings and the number of reviews to define the top books.

File: config/initializers/kaminari_config.rb (hunk 3 above)
Comment (taylorrf) [inline config/initializers/kaminari_config.rb]: I changed it for a better page visualisation after completed the design of the page.

File: spec/views/books/show.html.erb_spec.rb (hunk 4 above)
Comment (taylorrf) [inline spec/views/books/show.html.erb_spec.rb]: Thanks.

File: spec/views/authors/show.html.erb_spec.rb (hunk 6 above)
Comment (taylorrf) [inline spec/views/authors/show.html.erb_spec.rb]: Thanks.

File: spec/views/authors/show.html.erb_spec.rb (hunk 7 above)
Comment (taylorrf) [inline spec/views/authors/show.html.erb_spec.rb]: Much better, thanks.

File: spec/views/authors/show.html.erb_spec.rb (hunk 5 above)
Comment (taylorrf) [inline spec/views/authors/show.html.erb_spec.rb]: Thanks.

**Comment by andyw8** [comment]
Let's please not merge PRs without a :+1:

**Pull request merged**